src/lib/math.js

```diff
@@ -314,6 +314,47 @@ const $builtinmodule = function (name) {
         throw new Sk.builtin.NotImplementedError("math.isqrt() is not yet implemented in Skulpt");
     };
 
+    function lcm(...args) {
+        function abs(n) {
+            return JSBI.lessThan(result, JSBI.__ZERO)
+                ? new Sk.builtin.int_(JSBI.unaryMinus(result))
+                : new Sk.builtin.int_(result);
+        }
+
+        const nargs = args.length;
+
+        // lcm() without arguments returns 1
+        if (nargs === 0) return new Sk.builtin.int_(1);
+
+        let result = Sk.misceval.asIndexOrThrow(args[0]);
+
+        if (nargs === 1) {
+            return abs(result);
+        }
+
+        let i;
+        for (i = 1; i < nargs; ++i) {
+            let argument = Sk.misceval.asIndexOrThrow(args[i]);
+
+            // If any of the arguments is zero, then the returned value is 0
+            if (argument === 0) return new Sk.builtin.int_(0);
+
+            if (typeof result === "number" && typeof argument === "number") {
+                result = (result / gcd(result, argument)) * argument;
+            } else {
+                result = JSBI.BigInt(result);
+                argument = JSBI.BigInt(argument);
+
+                result = JSBI.multiply(
+                    JSBI.divide(result, gcd(result, argument)),
+                    argument
+                );
+            }
```

Comment:
gcd really shouldn't work here - 
gcd returns a python object and the fact that division is working here is really just accidental.
(JS converts the python objects to strings which and then tries to do division 🤷 )

I'd use the `_gcd_internal` function you suggested but returning a javascript object so that the division works like it should.


Another thing we should account for is the multiplication getting too large which means we have to stop working with  numbers and start working with bigints.

Something like this will probably work:

```suggestion
            if (typeof result === "number" && typeof argument === "number") {
                let tmp = (result / _gcd_internal(result, argument)) * argument;
                tmp = Math.abs(tmp);
                // check the size - if we're too big reset the result and then fall through
                result = tmp > Number.MAX_SAFE_INTEGER ? JSBI.BigInt(result) : tmp;
            } else {
                result = JSBI.BigInt(result);
            }

            // allow fall through - if result gets too big we'll need to redo the calculation with BigInts
            if (typeof result !== "number") {
                argument = JSBI.BigInt(argument);
                result = JSBI.multiply(JSBI.divide(result, _gcd_internal(result, argument)), argument);
            }
```